### PR TITLE
remove alloca

### PR DESCRIPTION
**Pull request by bramtayl**
Because alloca isn't available on all platforms

### CODE REVIEW

**Review by stekyne — APPROVED** [review]

**Comment by bramtayl** [comment]
bump

File: Engine/envvar.c [hunk 1]
```diff
@@ -1030,12 +1030,13 @@ void *csoundFileOpenWithType(CSOUND *csound, void *fd, int type,
 #if defined(WIN32)
       /* To handle Widows errors in file name characters. */
       size_t sz = 2 * MultiByteToWideChar(CP_UTF8, 0, name, -1, NULL, 0);
-      wchar_t *wfname = alloca(sz);
+      // alloca not available on all platforms
```
Comment (kunstmusik) [inline Engine/envvar.c:1033]: Don't really need this comment in code if we're removing alloca.

File: Engine/envvar.c [hunk 2]
```diff
@@ -1046,6 +1047,11 @@ void *csoundFileOpenWithType(CSOUND *csound, void *fd, int type,
         }
         fullName = (char*) name;
       }
+      free(wfname);
```
Comment (kunstmusik) [inline Engine/envvar.c:1050]: Because of the goto in line 1045, it's possible for wfname and wmode to leak memory.

**Review by kunstmusik — CHANGES_REQUESTED** [review]

File: Engine/envvar.c (hunk 2 above)
Comment (bramtayl) [reply Engine/envvar.c:1050]: Oh yeah oops. Would it be good enough to just copy the two free lines immediately before the goto?

File: Engine/envvar.c (hunk 1 above)
Comment (bramtayl) [reply Engine/envvar.c:1033]: The comment was mostly just so alloca doesn't sneak its way back in, but I can remove it too if you want

File: Engine/envvar.c (hunk 2 above)
Comment (bramtayl) [reply Engine/envvar.c:1050]: Also, is there a way to add a test for the memory leak?

File: Engine/envvar.c (hunk 1 above)
Comment (bramtayl) [reply Engine/envvar.c]: Ok, well, removed the comment

File: Engine/envvar.c (hunk 2 above)
Comment (bramtayl) [reply Engine/envvar.c:1051]: Ok, copied the frees before the goto

**Pull request merged**